Library/Homebrew/extend/pathname.rb

```diff
@@ -200,7 +200,7 @@ def extname(path = to_s)
     return archive_ext if archive_ext
 
     # Don't treat version numbers as extname.
-    return "" if basename.match?(/\b\d+\.\d+[^\.]*\Z/)
+    return "" if basename.match?(/\b\d+\.\d+[^\.]*\Z/) && !basename.end_with?(".7z")
```

Comment:
Could this be part of the regex?

Reply:
Maybe, but not in a readable way I assume.

Reply:
I [agree with @colindean](https://github.com/Homebrew/brew/pull/5000#pullrequestreview-160122055) regarding readability of this one.